src/db.cpp

```diff
@@ -66,7 +66,8 @@ PlainTextDb::PlainTextDb(const std::string& filename_,
 
 PlainTextDb::PlainTextDb(const std::string& filename_, bool is_system)
     : filename(filename_),
-      lock_file(LockFile::Get(LockFilePath(filename_).c_str())),
+      lock_file(LockFile::Get((is_system) ? LockFilePath(filename_).c_str()
+                                          : std::string(filename_ + ".lock").c_str())),
```

Comment:
Although I would advise changing LockFilePath so it would receive is_system and make the actual path.